airbyte-integrations/connectors/source-gcs/source_gcs/source.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import json
+from datetime import datetime
+from typing import Dict, Generator
+
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStream,
+    ConfiguredAirbyteCatalog,
+    Status,
+    Type,
+)
+from airbyte_cdk.sources import Source
+
+from .helpers import construct_file_schema, get_gcs_blobs, read_csv_file, get_stream_name
+MIN_BYTES_READ_GENERATE_SCHEMA = 102400
+
+
+class SourceGCS(Source):
+    def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
+        """
+        Check to see if a client can be created and list the files in the bucket.
+        """
+        try:
+            blobs = get_gcs_blobs(config)
+            if not blobs:
+                return AirbyteConnectionStatus(status=Status.FAILED, message="No compatible file found in bucket")
+            return AirbyteConnectionStatus(status=Status.SUCCEEDED)
+        except Exception as e:
+            return AirbyteConnectionStatus(status=Status.FAILED, message=f"An exception occurred: {str(e)}")
+
+    def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
+        streams = []
+
+        blobs = get_gcs_blobs(config)
+        for blob in blobs:
+            # Read the first 0.1MB of the file to determine schema
+            df = read_csv_file(
+                blob, limit_bytes=MIN_BYTES_READ_GENERATE_SCHEMA)
```

Comment:
why do you need to read 0.1 MB if you declare all columns to be strings? I believe reading just a single line would be enough to get the column names

Reply:
I don't think the gcs api allows us to read a single line of a CSV. Therefore I was thinking 0.1 MB should be enough to include all column names

Reply:
According to this [doc](https://cloud.google.com/storage/docs/streaming-downloads#storage-stream-download-object-python), the client library exposes a `download_blob_to_stream` method which returns a file like object. So after you get the file like object you can do something like 
```
for line in file_like_obj:
  first_line = line
  break
```